### PR TITLE
Py3k

### PR DESCRIPTION
**Pull request by jooddang**
test

### CODE REVIEW

File: nd_service_registry/funcs_tests.py [hunk 1]
```diff
@@ -47,8 +53,8 @@ def test_decode_returns_string_value_dict_on_malformed_json_string_input(
 
     def test_default_data_produces_expected_dict(self):
         default_data = funcs.default_data()
-        self.assertIn(u"pid", default_data.keys())
-        self.assertIn(u"created", default_data.keys())
+        self.assertIn(u"pid", list(default_data.keys()))
```
Comment (ashanbrown) [inline nd_service_registry/funcs_tests.py]: I think `list(default_data)` is sufficient.

File: nd_service_registry/registration.py [hunk 2]
```diff
@@ -395,5 +396,5 @@ def _update(self, data):
         # If the only key left in the self._data object is 'string_value',
         # then the supplied user data was actually in string format -- so
         # thats actually what we want to save.
-        if self._data.keys() == ['string_value']:
+        if list(self._data.keys()) == ['string_value']:
```
Comment (ashanbrown) [inline nd_service_registry/registration.py]: Again, list(self._data) should work.

File: nd_service_registry/registration.py [hunk 3]
```diff
@@ -238,7 +239,7 @@ def _create_node_path(self):
         # final path that will hold our node registration will get created
         # with whatever ACL settings were used when creating the Kazoo
         # connection object.
-        if len(filter(None, path.split('/'))) > 1:
+        if len([_f for _f in path.split('/') if _f]) > 1:
```
Comment (mrterry) [inline nd_service_registry/registration.py]: no need to change this, but you can spell this as: `if any(_f for _f in path.split('/') if _f)` too.

File: nd_service_registry/registration.py (hunk 2 above)
Comment (mrterry) [inline nd_service_registry/registration.py]: all things being equal, if i'm turning the keys of a dict into a list, i prefer calling `.keys()` explicitly.  it helps me remember what I'm working with.

File: requirements.txt [hunk 4]
```diff
@@ -1,4 +1,5 @@
-kazoo>=1.3
-setuptools
-python-gflags
+future
+kazoo==2.2
```
Comment (mrterry) [inline requirements.txt]: revert this to `kazoo>=1.3` please.  if the tests fail, we can reconsider.

File: nd_service_registry/registration.py (hunk 3 above)
Comment (jooddang) [inline nd_service_registry/registration.py]: @mrterry the len() should be greater than 1. If we do `any()` if returns `True` for len of 1.

**Comment by mrterry** [comment]
looks good to me.  @diranged can you merge this?  we will also need to push a new version.  you want to take point on that or give me privs?

**Comment by mikhail** [comment]
This looks great. I'm comfortable merging it, but not on Friday. I'll circle back on 2/21

**Pull request merged**